hpx/runtime/agas/big_boot_barrier.hpp

```diff
@@ -125,7 +125,8 @@ struct HPX_EXPORT big_boot_barrier : boost::noncopyable
 
     void add_thunk(util::unique_function_nonser<void()>* f)
     {
-        thunks.push(f);
+        while(!thunks.push(f))
+            ; // Wait until succesfully pushed ...
```

Comment:
Am 01.12.2015 4:48 nachm. schrieb "Hartmut Kaiser" <notifications@github.com
>:
>
> In hpx/runtime/agas/big_boot_barrier.hpp:
>
> > @@ -125,7 +125,8 @@ struct HPX_EXPORT big_boot_barrier :
boost::noncopyable
> >
> >      void add_thunk(util::unique_function_nonser<void()>* f)
> >      {
> > -        thunks.push(f);
> > +        while(!thunks.push(f))
> > +            ; // Wait until succesfully pushed ...
>
> This should time out with an error and not sit indefinitely.

push is guaranteed to succeed eventually, iiuc. Should we really time out
and throw here?

>
> —
> Reply to this email directly or view it on GitHub.